src/WebJobs.Script.WebHost/Diagnostics/DiagnosticLogger.cs

```diff
@@ -0,0 +1,96 @@
+            _eventGenerator = eventGenerator ?? throw new ArgumentNullException(nameof(eventGenerator));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _scopeProvider = scopeProvider ?? throw new ArgumentNullException(nameof(scopeProvider));
+
+            _regionName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.RegionName) ?? string.Empty;
+            _websiteHostName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName);
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => _scopeProvider.Push(state);
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            string formattedMessage = formatter?.Invoke(state, exception);
+
+            // Make sure we have something to log
+            if (string.IsNullOrEmpty(formattedMessage) && exception == null)
+            {
+                return;
+            }
+
+            (string exceptionType, string exceptionMessage, string exceptionDetails) = exception.GetExceptionDetails();
+
+            var scopeProps = _scopeProvider.GetScopeDictionary();
+
+            // Build up a JSON string for the Azure Monitor 'properties' bag
+            StringWriter sw = new StringWriter();
+            using (JsonTextWriter writer = new JsonTextWriter(sw) { Formatting = Formatting.None })
+            {
+                writer.WriteStartObject();
+                WritePropertyIfNotNull(writer, "message", Sanitizer.Sanitize(formattedMessage));
+                WritePropertyIfNotNull(writer, "category", _category);
+                WritePropertyIfNotNull(writer, "hostVersion", _hostVersion);
+                WritePropertyIfNotNull(writer, "functionInvocationId", Utility.GetValueFromScope(scopeProps, ScopeKeys.FunctionInvocationId));
+                WritePropertyIfNotNull(writer, "functionName", Utility.GetValueFromScope(scopeProps, ScopeKeys.FunctionName));
+                WritePropertyIfNotNull(writer, "hostInstanceId", _hostInstanceId);
+                WritePropertyIfNotNull(writer, "activityId", Utility.GetValueFromScope(scopeProps, ScriptConstants.LogPropertyActivityIdKey));
+                WritePropertyIfNotNull(writer, "level", logLevel.ToString());
+                WritePropertyIfNotNull(writer, nameof(exceptionDetails), exceptionDetails);
+                WritePropertyIfNotNull(writer, nameof(exceptionMessage), exceptionMessage);
+                WritePropertyIfNotNull(writer, nameof(exceptionType), exceptionType);
+                writer.WriteEndObject();
+            }
```

Comment:
Doing this the recommended way for best perf: https://www.newtonsoft.com/json/help/html/Performance.htm#ManuallySerialize